src/historywork/PutSnapshotFilesWork.cpp

```diff
@@ -124,4 +124,27 @@ PutSnapshotFilesWork::getFilesToZip()
 
     return filesToZip;
 }
+
+std::string
+PutSnapshotFilesWork::getStatus() const
+{
+    if (WorkUtils::getWorkStatus(mUploadSeqs) == State::WORK_RUNNING)
```

Comment:
I don't think this status check is needed here - you can simply check whether the vector is empty or not. empty means we haven't gotten to that step yet, non-empty means the step is running.

Reply:
fixed